### ISSUE TITLE
2.1.1 and 2.1.2 not following semantic versioning

### ISSUE THREAD

**Issue by marc1706** [s9e/SweetDOM#2]
I've noticed that both 2.1.1 and 2.1.2 are not following semantic versioning by breaking compatibility with PHP versions prior to PHP 8.0.
Would it be possible to have a version without this breaking change that is still compatible with PHP before 8.0?
Ideally all patch level releases in the 2.1 branch should be compatible with PHP 7.1 (as was the previous requirement) and later major or minor versions could potentially introduce this BC break.

Were there maybe other reasons that required this breaking change?

**Comment by JoshyPHP** [comment]
As far as I can remember, the library's API is backward compatible; That's why I didn't increase its major version nor do I consider it a BC break, but I will definitely consider bumping the version number more frequently in the future.

I don't remember anything in 2.1 that specifically requires PHP 8 but I haven't had access to PHP 7.1 for a while so I wouldn't be able to verify it. I have managed to compile some minimal version of PHP 7.1 (without ICU or SSL support) and I see at least one instance of a `match` construct. Downgrading to PHP 7.1 also means it would require re-adding a polyfill for DOM methods that were added in PHP 8. Effectively it would be reverting to 2.1.0.

Your best option may be to limit your requirements to 2.1.0 since the API is the same and only the implementation changes as far as I can tell. One caveat is that 2.1.2 contains a compatibility fix for PHP 8.3, but I would consider releasing a 2.1.0.1 version with that fix if that helps.
